tests/unit/utils/test_app_utils.py

```diff
@@ -78,6 +79,38 @@ async def test_application_upgrade_packages_with_hold(model):
     model.run_on_unit.assert_has_awaits(expected_calls)
 
 
+@pytest.mark.asyncio
+async def test_get_instance_count(model):
+    expected_count = 1
+    model.run_action.return_value = mocked_action = AsyncMock(spec_set=Action).return_value
+    mocked_action.results = {"Code": "0", "instance-count": str(expected_count)}
+
+    actual_count = await app_utils.get_instance_count(unit="nova-compute/0", model=model)
+
+    model.run_action.assert_called_once_with(
+        unit_name="nova-compute/0",
+        action_name="instance-count",
+    )
+    assert actual_count == expected_count
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "result_key, value",
+    [
+        ("not_valid", "1"),  # invalid key
+        ("instance-count", "not_valid"),  # invalid value
+        ("not_valid", "not_valid"), # invalid key and value
+    ],
+)
+async def test_get_instance_count_invalid_result(model, result_key, value):
+    model.run_action.return_value = mocked_action = AsyncMock(spec_set=Action).return_value
+    mocked_action.results = {"Code": "0", result_key: value}
+
+    with pytest.raises(ValueError):
+        await app_utils.get_instance_count(unit="nova-compute/0", model=model)
+
+
```

Comment:
Does it worth adding a test that shows the behavior if we pass a unit different than `nova-compute` to `.get_instance_count`

Reply:
Probably not at this stage since we are using it only internally. We didn't have unit checks for other helper functions either.

Reply:
The `model.run_action` will raise `ActionFailed`, so we do not need to test more.